src/types/plugin-input.ts

```diff
@@ -13,6 +13,10 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
 
 const ONE_DAY = 24 * 60 * 60 * 1000;
 
+const userRoleSchema = T.String();
```

Comment:
User role should be a `T.Literal` most likely

Reply:
@gentlementlegen are you suggesting that for this iteration @jordan-ae should define each pre-defined role as you say which was similar to how he had originally and any future custom role logic can refactor things later but for now be explicit and define each role?

Reply:
Maybe the best choice would be to have a `T.Union` with `T.Literal` and `T.String`. The idea is to have a better auto-complete and less error prone type.